### PR TITLE
[Arm64] ASIMD InsertScalar and rename to ShiftLeftAndInsert ShiftRightAndInsert 

### LINKED ISSUE #38137: [Arm64] ASIMD InsertScalar
These were omitted from #35037

```c#
class AdvSimd.Arm64
{
  // float64x2_t vcopyq_lane_f64 (float64x2_t a, const int lane1, float64x1_t b, const int lane2)
  //   A64: INS Vd.D[lane1], Vn.D[0]
  public static Vector128<double> InsertScalar(Vector128<double> result, byte resultIndex, Vector64<double> value);

  // int64x2_t vcopyq_lane_s64 (int64x2_t a, const int lane1, int64x1_t b, const int lane2)
  //   A64: INS Vd.D[lane1], Vn.D[0]
  public static Vector128<long> InsertScalar(Vector128<long> result, byte resultIndex, Vector64<long> value);

  // uint64x2_t vcopyq_lane_u64 (uint64x2_t a, const int lane1, uint64x1_t b, const int lane2)
  //   A64: INS Vd.D[lane1], Vn.D[0]
  public static Vector128<ulong> InsertScalar(Vector128<ulong> result, byte resultIndex, Vector64<ulong> value);
}
```

**Motivation:**
These would allow to implement **Vector128.WithLower()** as
```c#
AdvSimd.Arm64.InsertScalar(vector.AsUInt64(), 0, value.AsUInt64()).As<ulong, T>();
```
and **Vector128.WithUpper()** as
```c#
AdvSimd.InsertScalar(vector.AsUInt64(), 1, value.AsUInt64()).As<ulong, T>();
```

cc @CarolEidt @tannergooding @TamarChristinaArm

### PR DESCRIPTION
**Pull request by echesakov**
* Implements InsertScalar and uses them to implement Vector128\<T\>.WithLower and Vector128\<T\>.WithUpper

* Renames ShiftLeftLogicalAndInsert to ShiftLeftAndInsert and ShiftRightLogicalAndInsert to ShiftRightAndInsert as per https://github.com/dotnet/runtime/pull/36818#issuecomment-649718814

Fixes #38137

I attached JitDisasm for PMI-d System.Private.CoreLib.dll for Vector128\<T\>.WithLower and Vector128\<T\>.WithUpper
[System.Private.CoreLib-PMI+JitDisasm.txt](https://github.com/dotnet/runtime/files/4860145/System.Private.CoreLib-JitDisasm.txt)

### CODE REVIEW

**Comment by echesakovMSFT** [comment]
@CarolEidt @tannergooding @TamarChristinaArm PTAL

cc @kunalspathak for changes in WithLower and WithUpper

File: src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp [hunk 1]
```diff
@@ -644,6 +644,27 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
...
+                }
+
+                HWIntrinsicImmOpHelper helper(this, intrin.op2, node);
+
+                for (helper.EmitBegin(); !helper.Done(); helper.EmitCaseEnd())
+                {
+                    const int elementIndex = helper.ImmValue();
+
+                    GetEmitter()->emitIns_R_R_I_I(ins, emitSize, targetReg, op3Reg, elementIndex, 0, opt);
+                }
```
Comment (kunalspathak) [inline src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp:664]: Can you point me to the test that would exercise this code path?

File: src/coreclr/tests/src/JIT/HardwareIntrinsics/Arm/AdvSimd/InsertScalar.Vector128.Double.1.cs [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+        }
+
+        public void RunReflectionScenario_UnsafeRead()
+        {
+            TestLibrary.TestFramework.BeginScenario(nameof(RunReflectionScenario_UnsafeRead));
+
+            var result = typeof(AdvSimd).GetMethod(nameof(AdvSimd.InsertScalar), new Type[] { typeof(Vector128<Double>), typeof(byte), typeof(Vector64<Double>) })
+                                     .Invoke(null, new object[] {
+                                        Unsafe.Read<Vector128<Double>>(_dataTable.inArray1Ptr),
+                                        ElementIndex,
```
Comment (kunalspathak) [inline src/coreclr/tests/src/JIT/HardwareIntrinsics/Arm/AdvSimd/InsertScalar.Vector128.Double.1.cs:289]: >ElementIndex [](start = 40, length = 12)

I believe this is the only test that would verify the table generation for index?

File: src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp (hunk 1 above)
Comment (echesakovMSFT) [reply src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp:664]: InsertScalar tests I added in this PR

Comment (kunalspathak) [reply src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp:664]: Sure, I meant to ask which portion of `InsertScalar` test. I am just curious.

File: src/coreclr/tests/src/JIT/HardwareIntrinsics/Arm/AdvSimd/InsertScalar.Vector128.Double.1.cs (hunk 2 above)
Comment (echesakovMSFT) [reply src/coreclr/tests/src/JIT/HardwareIntrinsics/Arm/AdvSimd/InsertScalar.Vector128.Double.1.cs:289]: Correct, table generation is verified in the test reflection scenarios

File: src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp (hunk 1 above)
Comment (echesakovMSFT) [reply src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp:664]: The for loop is triggered in both cases - non-const imm and const imm. The difference is how many iterations is done and whether we insert any "header" the instruction(s) emitted for the intrinsic.

Say, for const case it's one iteration and there is no header. While for non const case we emit code that computes branching target given the non const value and emit multiple instructions (for each immValue in [immLowerBound, immUpperBound]) follower by `j endLabel`

Comment (kunalspathak) [reply src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp:664]: Ok, thanks!

**Review by kunalspathak — APPROVED** [review]
LGTM

**Review by tannergooding — APPROVED** [review]

**Pull request merged**